Kudu.Services/Functions/FunctionController.cs

```diff
@@ -22,6 +25,100 @@ public FunctionController(IFunctionManager manager, ITraceFactory traceFactory)
             _traceFactory = traceFactory;
         }
 
+        [HttpPut]
+        public async Task<HttpResponseMessage> CreateOrUpdate(string name)
+        {
+            var tracer = _traceFactory.GetTracer();
+            using (tracer.Step($"FunctionsController.CreateOrUpdate({name})"))
+            {
+                try
+                {
+                    var functionEnvelope = await Request.Content.ReadAsAsync<FunctionEnvelope>();
```

Comment:
Might consider allowing WebApi handle the content deserialization for you. Example [here](https://github.com/Azure/azure-webjobs-sdk-script/blob/master/src/WebJobs.Script.WebHost/Controllers/AdminController.cs#L33).